src/nats.rs

```diff
@@ -0,0 +1,26 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields, rename_all = "snake_case", tag = "strategy")]
+pub(crate) enum NatsAuthConfig {
+    UserPassword { user: String, password: String },
+    Token { token: String },
+    CredentialsFile { path: String },
+    NKey { nkey: String, seed: String },
+}
+
+impl NatsAuthConfig {
+    pub(crate) fn to_nats_options(&self) -> async_nats::Options {
```

Comment:
This could maybe be a `impl From<NatsAuthConfig> for async_nats::Options` (or `TryFrom` if it could fail)